glide-core/telemetry/src/open_telemetry.rs

```diff
@@ -0,0 +1,240 @@
+use opentelemetry::global::ObjectSafeSpan;
+use opentelemetry::trace::SpanKind;
+use opentelemetry::{global, trace::Tracer};
+use opentelemetry_sdk::propagation::TraceContextPropagator;
+use opentelemetry_sdk::trace::TracerProvider;
+use std::path::PathBuf;
+use std::sync::{Arc, RwLock};
+
+const SPAN_WRITE_LOCK_ERR: &str = "Failed to get span write lock";
+const TRACE_SCOPE: &str = "valkey_glide";
+
+pub enum GlideSpanStatus {
+    Ok,
+    Error(String),
+}
+
+#[allow(dead_code)]
+#[derive(Clone, Debug)]
+/// Defines the tracing exporter between GLIDE -> collector
```

Comment:
add more explanations about the exporter. something like: To optimize performance, the exporter often batches spans or metrics together instead of sending each piece of telemetry data individually.